cpp/src/ray/util/function_helper.h

```diff
@@ -1,39 +1,67 @@
 #pragma once
 
 #include <ray/api/common_types.h>
+#include <ray/api/function_manager.h>
 #include <boost/dll.hpp>
 #include <memory>
 #include <msgpack.hpp>
 #include <string>
 #include <unordered_map>
 
+using namespace ::ray::internal;
+
 namespace ray {
 namespace api {
 
+using EntryFuntion = std::function<msgpack::sbuffer(const void *, const std::string &,
+                                                    const std::vector<msgpack::sbuffer> &,
+                                                    msgpack::sbuffer *)>;
+
 class FunctionHelper {
  public:
   static FunctionHelper &GetInstance() {
     static FunctionHelper functionHelper;
     return functionHelper;
   }
 
-  std::shared_ptr<boost::dll::shared_library> LoadDll(const std::string &lib_name);
-  std::function<msgpack::sbuffer(
-      const std::string &, const std::vector<msgpack::sbuffer> &, msgpack::sbuffer *)>
-  GetEntryFunction(const std::string &lib_name);
+  void LoadDll(const boost::filesystem::path &lib_path);
+  void LoadFunctionsFromPaths(const std::vector<std::string> paths);
+  template <typename F>
+  std::pair<EntryFuntion, const F *> LookupExecutableFunctions(
+      const std::string &function_name, std::string lib_path,
+      const std::unordered_map<std::string, F> &funcs_map) {
+    EntryFuntion entry_function;
+    // Lookup function pointer.
+    auto it = funcs_map.find(function_name);
+    if (it == funcs_map.end()) {
+      return std::make_pair(entry_function, nullptr);
```

Comment:
Why can't find the function name? i think we register all functions, if we can't find the function, it should be an invalid request, and we need to handle the exception, don't return nullptr.

Reply:
If you has multiple dynamic library, some map finding will return `nullptr`.

Reply:
And I have throw an exception here https://github.com/ray-project/ray/pull/16828/files#diff-d7517919ec685a9ab470810f835ec5912fae4699db5d5494223a5278b4e9caf4R160.